torch/autograd/gradcheck.py

```diff
@@ -401,6 +445,29 @@ def not_reentrant_error(error_str=''):
         if out_is_complex and not reentrant_with_imag_grad_out:
             return not_reentrant_error(' (calculated using complex valued grad output)')
 
+        if check_forward:
+            try:
+                fw_analytical = get_analytical_jacobian_fw(fn, tupled_inputs, o)
+            except RuntimeError as e:
+                msg = str(e)
+                # If the jvp formula isn't implemented, then we will warn the user. Otherwise, if the formula is
+                # implemented, then we check the result for correctness
+                if "Trying to use forward prop with" in msg:
+                    warnings.warn("Failed to compute gradcheck using fw mode: {}".format(msg), stacklevel=stacklevel)
+                    continue
+                else:
+                    raise e
+
+            if fw_analytical is None:
+                # Test was aborted while computing the jacobian
+                continue
+
+            for j, (a, n) in enumerate(zip(fw_analytical, numerical)):
+                if a.numel() != 0 or n.numel() != 0:
+                    if not torch.allclose(a, n, rtol, atol):
+                        return fail_test('Jacobian mismatch for output %d with respect to input %d,\n'
+                                         'numerical:%s\nforward analytical:%s\n' % (i, j, n, a))
```

Comment:
nit: continues can reduce the indentation, but it probably comes down to personal preference.
```
if a.numel() == 0 and n.numel() == 0:
    continue
if torch.allclose(a, n, rtol, atol):
    continue
return fail_test(...)
```

Reply:
Also, if a.numel() == 0 and n.numel() == 0, shouldn't we check that the shapes of the tensors are the same?